tensorwaves/data/generate.py

```diff
@@ -45,7 +46,7 @@ def _generate_data_bunch(
 
 def generate_data(
     size: int,
-    kinematics: Kinematics,
+    kinematics: HelicityKinematics,
```

Comment:
Hmm, small comment. Originally the idea was to have this generate function work generally, independent on the Kinematics. However the `PhaseSpaceGenerator`s are working based on particle transition info.
I was thinking about duck typing here. That the specific Kinematics instance would have to define this attribute for the generate function to work w/o errors... Is a bit of a code smell though. But I also wanted the interface to be simple. Maybe you have a good idea for this problem.

So I guess for now we can go with `HelicityKinematics`, and create some Issue for this general Kinematics option.

Reply:
Yes, that's true and I understood this was done with duck typing in mind. Did make me wonder what typing and mypy thinks about duck typing. I'm inclined to avoid duck typing in combination with type hinting.

As for our design: I initially gave a try adding the transition info into the `Kinematics` base class, but that changed the design too radically.